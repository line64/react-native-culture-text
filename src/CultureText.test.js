import 'react-native';
import React from 'react';
import { View } from 'react-native';
import { CultureContext, Text } from '../';

import renderer from 'react-test-renderer';

const translations = {
  es: {
    welcome: 'Este texto depende de la cultura seleccionada',
  },
}

it('render translated message', () => {
  const tree = renderer.create(
    <CultureContext translations={translations} locale="es">
      <View>
        <Text>
          Welcome to React Native!
        </Text>
        <Text messageKey="welcome">
          This text depends on the selected culture
        </Text>
      </View>
    </CultureContext>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
