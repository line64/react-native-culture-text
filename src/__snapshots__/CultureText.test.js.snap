// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render translated message 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Este texto depende de la cultura seleccionada
  </Text>
</View>
`;
